server/plugin.go

```diff
@@ -204,6 +208,91 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	go p.initStats()
+
+	go func() {
+		time.Sleep(time.Second * 10)
+
+		instances, err := p.instanceStore.LoadKnownJIRAInstances()
+		if err != nil {
+			p.API.LogError("unable to register autolinks", "err", err)
+			return
+		}
+
+		for url := range instances {
+			instance, err := p.instanceStore.LoadJIRAInstance(url)
+			if err != nil {
+				continue
+			}
+
+			jci, ok := instance.(*jiraCloudInstance)
```

Comment:
This is ok for the initial implementation, but as far as I understand, there's no reason for us not to do this for Jira Server. Should a separate ticket be created for that?

Reply:
I was hoping to do it as part of this PR. Looking for guidance on how to do that since there doesn't seem to be existing methods to get a generic client for a server instance. (I feel like I am just missing something)

Reply:
> there doesn't seem to be existing methods to get a generic client for a server instance

I don't believe there is a way we can communicate to Jira server, unless we have a dedicated bot user in the Jira instance. Every API call seems to need to be tied to a user https://developer.atlassian.com/server/jira/platform/security-overview

Reply:
If we have to have the user to get the list of projects, we should defer, but file an (HW?) ticket to add the capability of mapping the "bot" user with a command, perhaps, or as a part of `/jira install server`. cc @aaronrothschild @jfrerich - seems like we need to spec it out and file soon.

Reply:
Follow up ticket for the bot mapping for server: https://mattermost.atlassian.net/browse/MM-22076

A good HW ticket for a temporary solution for server instances: https://mattermost.atlassian.net/browse/MM-22077